nimbus/src/enrollment.rs

```diff
@@ -690,6 +720,30 @@ pub fn opt_out(db: &Database, experiment_slug: &str) -> Result<Vec<EnrollmentCha
     Ok(events)
 }
 
+/// Reset unique identifiers in response to application-level telemetry opt-out.
+///
+/// Each experiment enrollment record contains a unique `enrollment_id`. When the user
+/// opts out of application-level telemetry, we reset each such id to a new random value,
+/// creating a clean break between data sent before the opt-out and any data they might
+/// send in future.
+///
+/// Note that we do *not* change their experiment enrollments, because we don't want
+/// the act of disabling telemetry to cascade into other unrelated changes in the app
+/// as the user moves out of active experiment branches.
```

Comment:
This raises an interesting concern if the user re-enables telemetry with a new randomization unit that no longer qualifies the client bucket for the experiment. I'm not quite sure what to do with the existing enrollments, and how do we know if the user enables telemetry again?

Reply:
Indeed. I'm wondering if we can move any existing enrolled experiments to `Disqualified` state to prevent weird client-side behaviours or data analysis artifacts if the user re-enables telemetry.

Reply:
Moving them to disqualified seems reasonable to me, along with resetting the enrollment_id to a canary value.